src/main/java/io/fabric8/jenkins/openshiftsync/BuildDecisionHandler.java

```diff
@@ -107,7 +109,22 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
         logger.info("Triggering OpenShift Build for WorkflowJob " + wj.getFullName() + " due to " + causeDescription(actions));
         BuildConfig foundBuildConfig = OpenShiftUtils.findBCbyNameOrLabel(getOpenShiftClient(), namespace, buildConfigname);
         if (foundBuildConfig != null) {
+          int buildInJenkins = wj.getLastBuild()!=null ? wj.getLastBuild().number : 0;
           buildConfigname = foundBuildConfig.getMetadata().getName();
+          if (isPRBuild && (buildInJenkins > 0 || (buildInJenkins == 0 && isJobInQueue(wj)))) {
+            if (isJobInQueue(wj)){
+              cancelBuildRequest(buildConfigname, namespace, wj.getNextBuildNumber());
+              Jenkins.getInstance().getQueue().cancel(wj);
```

Comment:
Was searching Jenkins for some usages of this and discovered that Jimmi had a version of this already in JenkinsUtils ... see https://github.com/fabric8io/jenkins-sync-plugin/blob/job-to-bc/src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java#L256-L292

Even if you need a slightly modified versions of those current foms, you should place whatever new flavor of the cancel queued build function you derive in JenkinsUtils.java, and leverage the same impersonate pattern applied there.
@gabemontero